winstore/Program.cs

```diff
@@ -0,0 +1,271 @@
+            StoreCode = storeCode;
+        }
+
+        /// <summary>
+        /// Replaces $VARIABLES$ within the specified string with translated strings.
+        /// </summary>
+        /// <param name="langPath">Path to the 'lang' folder.</param>
+        /// <param name="stringToSubstitute">String to work on.</param>
+        /// <returns>The substituted string.</returns>
+        public string Substitute(string langPath, string stringToSubstitute)
+        {
+            if (strings == null)
+            {
+                strings = new Dictionary<string, string>(StringComparer.Ordinal);
+                string path = Path.Combine(langPath, Path.ChangeExtension(LangFile, ".txt"));
+
+                if (!File.Exists(path))
+                {
+                    // TODO: For now, substitute the English version.
+                    path = Path.Combine(langPath, "english.txt");
+                }
+
+                using (FileStream fs = File.Open(path, FileMode.Open, FileAccess.Read, FileShare.Read))
+                {
+                    using (StreamReader sr = new StreamReader(fs, Encoding.UTF8))
+                    {
+                        while (!sr.EndOfStream)
+                        {
+                            string? line = sr.ReadLine();
+
+                            if (string.IsNullOrWhiteSpace(line) || line.StartsWith('#'))
+                            {
+                                continue;
+                            }
+
+                            int firstColon = line.IndexOf(':');
+
+                            if (firstColon < 0)
+                            {
+                                continue;
+                            }
+
+                            string key = $"${line.Substring(0, firstColon).Trim()}$";
+                            string value = line.Substring(firstColon + 1).Trim();
+
+                            strings[key] = value;
+                        }
+                    }
+                }
+            }
+
+            foreach (KeyValuePair<string, string> kvp in strings)
+            {
+                stringToSubstitute = stringToSubstitute.Replace(kvp.Key, kvp.Value, StringComparison.Ordinal);
+            }
+
+            return stringToSubstitute;
+        }
+    }
+
+    internal class Program
+    {
+        /// <summary>
+        /// List of supported languages.
+        /// </summary>
+        private static readonly List<Language> languageMappings =
+        [
+            new Language("arabic_egypt", "ar-eg"),
+            new Language("bulgarian", "bg-bg"),
+            new Language("simplified_chinese", "zh-cn"),
+            new Language("traditional_chinese", "zh-tw"),
+            new Language("bulgarian", "bg-bg"),
+            new Language("czech", "cs-cz"),
+            new Language("danish", "da-dk"),
+            new Language("dutch", "nl-nl"),
+            new Language("finnish", "fi-fi"), // Not currently selected for OpenTTD on the Windows Store
```

Comment:
If that comment is true, why is it in here? Or might the comment become untrue at some point? Or should we disable the line in total? :)

Reply:
It's more a "todo" comment for me; once we have translations running through this, I'll add it to the WinStore listing. I'll remove the comment now.